config/append.go

```diff
@@ -17,7 +17,7 @@ package config
 import (
 	"reflect"
 
-	"github.com/coreos/ignition/internal/config/types"
+	"github.com/coreos/ignition/config/v3_0_experimental/types"
```

Comment:
In the case where the imported name is `types` the maintenance shouldn't be too bad, just changing the import names. In places where we import the config package, I explicit import it as `config` not `v3_0_experimental`.

Using the internal types is problematic right now since we need to define functions like `Key()` on them and we lose that when translating from `vX_Y` -> `internal`.

So basically, yes, does make config bumps slightly harder, but IMO it's worth it.